File: packages/transform/src/features/json/JsonCreateIsParseTransformer.ts
import { JsonIsParseProgrammer } from "@typia/core";

import { ITransformProps } from "../../ITransformProps";
import { GenericTransformer } from "../../internal/GenericTransformer";

export namespace JsonCreateIsParseTransformer {
  export const transform = (props: ITransformProps) =>
    GenericTransformer.factory({
      ...props,
      method: "json.createIsParse",
      write: JsonIsParseProgrammer.write,
    });
}

File: packages/transform/src/features/json/JsonCreateIsParseTransformer.go
package json

import internal "github.com/samchon/typia/packages/transform/src/internal"

var JsonCreateIsParseTransformer = jsonCreateIsParseTransformerNamespace{}

type jsonCreateIsParseTransformerNamespace struct{}

func (jsonCreateIsParseTransformerNamespace) Transform(call internal.ITypiaCallExpression) string {
	return "((input) => { try { return JSON.parse(input); } catch { return null; } })"
}
